cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java

```diff
@@ -133,10 +139,10 @@ public void putOAuthCredential(HttpServiceRequest request, HttpServiceResponder
       try {
         putOAuthCredentialRequest = GSON.fromJson(StandardCharsets.UTF_8.decode(request.getContent()).toString(),
                 PutOAuthCredentialRequest.class);
-        if (putOAuthCredentialRequest.getOneTimeCode().isEmpty()) {
+        if (putOAuthCredentialRequest.getOneTimeCode() == null) {
```

Comment:
Does it makes sense to check for both null and empty? This would be safer to handle `oneTimeCode=""` kind of json. I also think we have commons-lang, so you can use `org.apache.commons.lang.StringUtils#isEmpty` here and in the return.

Reply:
Done, thanks